webclient/webclient/src/main/java/io/helidon/webclient/NettyClientHandler.java

```diff
@@ -273,27 +279,44 @@ public int emit(final ByteBuf buf) {
 
     final class ResponseCloser {
 
-        private final AtomicBoolean closed;
+        private final AtomicReference<Single<Void>> singleClosed;
         private final ChannelHandlerContext ctx;
+        private final CompletableFuture<Void> cf;
 
         ResponseCloser(ChannelHandlerContext ctx) {
             this.ctx = ctx;
-            this.closed = new AtomicBoolean();
+            this.singleClosed = new AtomicReference<>();
+            cf = new CompletableFuture<>();
         }
 
         boolean isClosed() {
-            return closed.get();
+            return singleClosed.get() != null;
         }
 
-        ChannelFuture close() {
-            if (closed.compareAndSet(false, true)) {
+        /**
+         * Asynchronous close method.
+         *
+         * @return single of the closing process
+         */
+        Single<Void> close() {
+            if (singleClosed.compareAndSet(null, Single.from(cf, true))) {
```

Comment:
This approach is not correct.
1. you will create a new `Single` every time method `close` is called
2. you should create a new single for each response
3. isClosed may use the `cf.isDone` rather than this strange construct